ssh-sk.c

```diff
@@ -530,7 +530,7 @@ sshsk_enroll(int type, const char *provider_path, const char *device,
 	}
 	/* XXX validate flags? */
 	/* enroll key */
-	if ((r = skp->sk_enroll(alg, challenge, challenge_len, application,
+	if ((r = skp->sk_enroll(alg, challenge, challenge_len, application, // CodeQL [SM02311]: false positive challenge will not be null
```

Comment:
Please add NULL check for `challenge`, because `sshbuf_ptr` can return NULL if sanity check fails.